microservices-connector/config/samples/ChatQnA/use_cases.md

```diff
@@ -0,0 +1,222 @@
+
+## Deploy ChatQnA pipeline
+
+There are 3 use cases for ChatQnA example:
+
+- General ChatQnA with preset RAG data
+- ChatQnA with data preparation which supports that the user can upload RAG data online via dataprep microservice
+- ChatQnA supports multiple LLM models which can be switched in runtime
+
+### General ChatQnA with preset RAG data
+
+This involves deploying the ChatQnA custom resource. You can use `chatQnA_xeon.yaml` or if you have a Gaudi cluster, you could use `chatQnA_gaudi.yaml`.
+
+1. Create namespace and deploy application
+
+   ```sh
+   kubectl create ns chatqa
+   kubectl apply -f $(pwd)/chatQnA_xeon.yaml
+   ```
+
+2. GMC will reconcile the ChatQnA custom resource and get all related components/services ready. Check if the service up.
+
+   ```sh
+   kubectl get service -n chatqa
+   ```
+
+3. Retrieve the application access URL
+
+   ```sh
+   kubectl get gmconnectors.gmc.opea.io -n chatqa
+   NAME     URL                                                      READY     AGE
+   chatqa   http://router-service.chatqa.svc.cluster.local:8080      9/0/9     3m
+   ```
+
+4. Deploy a client pod to test the application
+
+   ```sh
+   kubectl create deployment client-test -n chatqa --image=python:3.8.13 -- sleep infinity
+   ```
+
+5. Access the application using the above URL from the client pod
+
+   ```sh
+   export CLIENT_POD=$(kubectl get pod -n chatqa -l app=client-test -o jsonpath={.items..metadata.name})
+   export accessUrl=$(kubectl get gmc -n chatqa -o jsonpath="{.items[?(@.metadata.name=='chatqa')].status.accessUrl}")
+   kubectl exec "$CLIENT_POD" -n chatqa -- curl -s --no-buffer $accessUrl  -X POST  -d '{"text":"What is the revenue of Nike in 2023?","parameters":{"max_new_tokens":17, "do_sample": true}}' -H 'Content-Type: application/json'
+   ```
+
+6. Perhaps you want to try another LLM model? Just modify the application custom resource to use another LLM model
+
+   Should you, for instance, want to change the LLM model you are using in the ChatQnA pipeline, just edit the custom resource file.
+   For example, to use Llama-2-7b-chat-hf make the following edit:
+
+   ```yaml
+   - name: Tgi
+     internalService:
+       serviceName: tgi-service-m
+       config:
+         LLM_MODEL_ID: Llama-2-7b-chat-hf
+   ```
+
+7. Apply the change
+
+   ```
+   kubectl apply -f $(pwd)/chatQnA_xeon.yaml
+   ```
+
+8. Check that the tgi-svc-deployment has been changed to use the new LLM Model
+
+   ```sh
+   kubectl get deployment tgi-service-m-deployment -n chatqa -o jsonpath="{.spec.template.spec.containers[*].env[?(@.name=='LLM_MODEL_ID')].value}"
+   ```
+
+9. Access the updated pipeline using the same URL from above using the client pod
+
+   ```sh
+   kubectl exec "$CLIENT_POD" -n chatqa -- curl -s --no-buffer $accessUrl -X POST -d '{"text":"What is the revenue of Nike in 2023?","parameters":{"max_new_tokens":17, "do_sample": true}}' -H 'Content-Type: application/json'
```

Comment:
Please double check the question. I'm not sure if the nke.pdf data is imported or not by default. If there is no nike data import, you should change a question here.

Reply:
Hi @yongfengdu, okay, I change the question as the same below.